internal/api/api.go

```diff
@@ -26,6 +28,14 @@ func (r RepoName) Equal(o RepoName) bool {
 	return strings.EqualFold(string(r), string(o))
 }
 
+var deletedRegex = lazyregexp.New("DELETED-[0-9]+\\.[0-9]+-")
+
+// UndeleteRepoName will "undeleted" a repo name. When we soft-delete a repo we
+// change its name in the database, this function extract the original repo name.
+func UndeleteRepoName(name RepoName) RepoName {
```

Comment:
Either the above, or maybe you wanted to go for `UndeletedRepoName`? 

`UndeleteRepoName` sounds like a more persistent state change vs `UndeletedRepoName`.  Maybe it's just me overthinking. :thinking: 

Feel free to ignore.

```suggestion
func UndeletedRepoName(name RepoName) RepoName {
```

Reply:
Agree that 'UndeletedRepoName' is better, I'll rename